proxy/internal/service/init.go

```diff
@@ -49,29 +50,77 @@ func initTrackers(p *Proxy) error {
 }
 
 func initIndex(p *Proxy) error {
-	var err error
-	indexPath := p.config.ModuleStorage.IndexPath
-	p.index, err = store.NewIndex(store.IndexOptions{
-		Directory: indexPath,
-	})
-	return err
+	if p.config.ModuleStorage != nil {
+		indexPath := p.config.ModuleStorage.IndexPath
+		index, err := store.NewIndex(store.IndexOptions{
+			Directory: indexPath,
+		})
+		if err != nil {
+			return errors.WithStack(err)
```

Comment:
why do we want the stack trace?

Reply:
context. It's either that or wrapping with an extra message (which will add the stack anyway), or returning a new error. I think this option is the easiest, while still giving the operator all the context she needs.